src/libraries/System.Text.Json/src/System/Text/Json/JsonPropertyDictionary.ValueList.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Text.Json
+{
+    internal sealed partial class JsonPropertyDictionary<T>
+    {
+#if DEBUG
+        // CreateEditableValueList should be called at most once by JsonTypeInfo
```

Comment:
Nit: "Editable" means "mutable" in this case?

Reply:
Yes, I didn't want to spend too much time on figuring out better name for internal class. I chose Editing here because we don't act on the original data structure and rather on its copy until we finish editing